duckduckgo/Android
# Add logic to provide system app overrides

Task/Issue URL: https://app.asana.com/0/488551667048375/1201403499648047/f

### Description
Some beta users have flagged that some of the popular apps that are installed in their device don't appear under the "Manage protection for your apps" screen. This is the case for eg. Chrome

These apps don't appear because they're flagged by Android as "system apps" as all system apps are unprotected but also don't show in the "Manage protection for your apps" screen:
* they don't appear in there because there's a lot of internal Android libraries are also system apps
* showing them all would clutter that screen and create confusion

However we need to add those that are popular apps so that users have control over their protection.


This PR uses a new endpoint to fetch the system app overrides and also modifies the logic so that overrided apps appear under the "Manage protection for your apps" screen

### Steps to test this PR
Install chrome browser

_test the feature_
* install app from develop
* enable appTP
* navigate to "Manage protection for your apps"
* Chrome should not appear in the list of apps
* install from this branch
* enable appTP
* navigate to "Manage protection for your apps"
* Chrome should appear in the list of apps and you can protect/unprotect

_test the migration_
This feature introduces for the first time migrations
1. install from develop
2. enable appTp
3. verify the vpn database does not contain neither the `vpn_app_tracker_system_app_override_list` nor the `vpn_app_tracker_system_app_override_list_metadata` tables
4. install/update from this branch
5. launch the app
6. verify app doesn't crash
7. verify app contains the `vpn_app_tracker_system_app_override_list` AND the `vpn_app_tracker_system_app_override_list_metadata` tables


The initial list of overrides for system apps are [here](https://github.com/duckduckgo/app-tracking-protection-lists/pull/21/files)
